packages/api/src/models/animalBatchModel.js

```diff
@@ -141,6 +144,19 @@ class AnimalBatchModel extends baseModel {
           to: 'animal_batch_use_relationship.animal_batch_id',
         },
       },
+      tasks: {
```

Comment:
Is it more efficient if we have already retrieved all tasks on frontend? I am worried about nesting base objects into other base objects like existing saga complexity. 

When we invalidate tags if we add a new animal task, we would need to invalidate both [Animals + Tasks] ?

Reply:
So when a user removes an animal, would tasks associated with that animal be included in the API request as well? It feels more natural and easier to handle it on the backend...

The cashing is a good point - if we were using RTK query for tasks, I would think that invalidating tasks might reflect changes on animals. But since animals wouldn't automatically receive new or updated tasks with the current set up, joining them in API responses doesn't seem like a good idea.

Reply:
Good point about removal cascade and agreed it should be handled on backend .

And I think for frontend I must have thought that we would be modifying tasks slice to receive the `animal_task_relationship` table -- to your earlier point about not returning relationships -- then its not a good idea... Okay I think I understand now 🙏